spec/concurrent/array_spec.rb

```diff
@@ -14,5 +14,15 @@ module Concurrent
         end
       end.map(&:join)
     end
+
+    describe '#slice' do
+      # This is mostly relevant on Rubinius and Truffle
```

Comment:
The problem is not so much that this method is public rather than that they are "special" to Rubinius in that they create objects without calling `allocate`. They would still have to be overwritten by concurrent-ruby if they were private, although it would have to be done slightly different...

As for the comment itself, I have not tested this on Truffle nor have I specifically checked their implementation to be honest. The comment might thus be not 100% exact in this regard. However, since the monitor is only used on Rubinius and Truffle at all, the test still applies to this locking variant used here.